src/core/include/openvino/core/type.hpp

```diff
@@ -78,37 +78,55 @@ OPENVINO_API
 std::ostream& operator<<(std::ostream& s, const DiscreteTypeInfo& info);
 
 #if defined(__ANDROID__) || defined(ANDROID)
-#    define OPENVINO_DYNAMIC_CAST
+inline constexpr auto use_ov_dynamic_cast = true;
+#else
+inline constexpr auto use_ov_dynamic_cast = false;
```

Comment:
```suggestion
inline constexpr bool use_ov_dynamic_cast{
#ifdef OPENVINO_DYNAMIC_CAST
    true
#endif 
};
```

Reply:
Would it be faster? ;)

Reply:
This is optional just less #ifdef.
Could be:
```inline constexpr bool use_ov_dynamic_cast{ OPENVINO_DYNAMIC_CAST };```
if `OPENVINO_DYNAMIC_CAST` would bool flag.

Consider put this variable in namespace e.g. `ov::compile::use_dynamic_cast` ?

Reply:
After merging #28834 it's irrelevant .. closing.